jobs/build-podman-os.Jenkinsfile

```diff
@@ -0,0 +1,347 @@
+                           }]}
+                       }
+                   }
+            }]}
+        }
+        withCredentials([file(credentialsId: 'ravanelli-push-registry-secret', variable: 'REGISTRY_SECRET')]) {
+            stage("Push Manifest") {
+
+                def manifest = "podman-v5"
+                def cmds = ""
+                parallel archinfo.keySet().collectEntries{arch -> [arch, {
+                    artifacts.each { artifact ->
+                        def file = "podman-$arch-${shortcommit}.${artifact["suffix"]}.zst"
+                        def platform = "${artifact["platform"]}"
+                        cmds += " && /usr/bin/buildah manifest add --artifact --artifact-type='' \
+                                --os=linux --arch=${arch} --annotation 'disktype=${platform}' ${manifest} ${file}"
+                    }
+                }]}
+                //  Supermin is unable to access the secret mounted in another directory.
+                // Although this solution is not ideal, let's proceed with it until we
+                // can find a more effective approach.
+                // This doc includes a 'use' case that didn't function as expected.
+                // https://www.jenkins.io/doc/pipeline/steps/credentials-binding/
+                shwrap("""cp ${REGISTRY_SECRET} push-secret""")
+
+                // We need to run all buildah commands in the same superminvm to not lose
+                // the manifest creation/additions
+                shwrap("""cosa supermin-run bash -c \
+                    "/usr/bin/buildah manifest create ${manifest} ${cmds} && \
+                    /usr/bin/buildah manifest push --authfile ./push-secret --all ${manifest} \
+                    docker://${params.CONTAINER_REGISTRY_STAGING_REPO}:5.0-${shortcommit}"
+                """)
+                shwrap("""rm -Rf  push-secret""")
+            }
+        }
+        if (params.MANIFEST_RELEASE) {
+            withCredentials([file(credentialsId: 'ravanelli-push-registry-secret', variable: 'REGISTRY_SECRET')]) {
+                stage("Release Manifest") {
+                    shwrap("""
+                        skopeo copy --all --authfile \$REGISTRY_SECRET   \
+                        docker://${params.CONTAINER_REGISTRY_STAGING_REPO}:5.0-${shortcommit} \
+                        docker://${params.CONTAINER_REGISTRY_REPO}:5.0
+                    """)
+                    parallel archinfo.keySet().collectEntries{arch -> [arch, {
+                        // Release the manifest and container images
+                        shwrap("""
+                            skopeo copy --all --authfile \$REGISTRY_SECRET   \
+                            docker://${params.CONTAINER_REGISTRY_STAGING_REPO}:${arch}-${shortcommit} \
+                            docker://${params.CONTAINER_REGISTRY_REPO}:5.0-${arch}
+                        """)
+                    }]}
```

Comment:
can you help me understand what is going on here behind the scenes?

What is the difference between these two?:
- staging to main repo `5.0-${shortcommit} -> 5.0`
- for each arch
    - staging to main repo `${arch}-${shortcommit} -> 5.0-${arch}`

Reply:
yes. I think the question I'm asking is: why?

why do we need that? are the users going to use them directly? is it some artifact of the way we are pushing that they need to be there?

Reply:
I think it is a question for @baude

Reply:
@dustymabe i dont get what you are asking me, ill reach out

Reply:
ok talked with brent,

basically we need the `5.0` tag to include the oci artifacts too (i.e. the containers and not just the disk images) so we'll need to add a few more steps to the `buildah manifest` command run in the supermin VM. 

we don't actually need the `5.0-aarch64` and `5.0-x86_64` in the registry so we can just not push those (i.e. we can delete the second `skopeo copy`).